src/lib.rs

```diff
@@ -301,6 +301,17 @@ impl<T> AtomicLazyCell<T> {
             _ => None,
         }
     }
+    /// Borrows the contents of this lazy cell mutably for the duration of the
+    /// cell itself.
+    ///
+    /// This function will return `Some` if the cell has been previously
+    /// initialized, and `None` if it has not yet been initialized.
+    pub fn borrow_mut(&mut self) -> Option<&mut T> {
+        match self.state.load(Ordering::Acquire) {
```

Comment:
Atomic op seems confusing here, perhaps get_mut will be more clear? https://doc.rust-lang.org/std/sync/atomic/struct.AtomicUsize.html#method.get_mut

Reply:
@matklad sorry to necrobump this PR, but could you expand on why the atomic op is confusing? I'm not very well acquainted with the atomic operations.

Reply:
It is confusing in a sense that we do atomic load when we don’t need one. We have &mut anyway, so additional synchronization power which comes from atomic load is unnecessary. Using extra power when there’s no need for it confuses me, as I think that there should be some real reason for it.